src/main/java/io/appform/jsonrules/expressions/JsonPathBasedExpression.java

```diff
@@ -55,30 +70,55 @@ protected JsonPathBasedExpression(ExpressionType type, String path, boolean defa
 
     @Override
     public final boolean evaluate(ExpressionEvaluationContext context) {
-        JsonNode evaluatedNode = applyPreoperation(context);
-
-        if (evaluatedNode.isMissingNode()) {
+        JsonNode nodeAtPath = null;
+        try {
+            val nodeValue = JsonPath.read(context.getNode().toString(), path);
+            if (nodeValue != null) {
+                nodeAtPath = mapper.valueToTree(nodeValue);
+            } else {
+                // Node exists; but value is null. Proceed as missing node.
+                nodeAtPath = MissingNode.getInstance();
+            }
+        } catch (PathNotFoundException exception) {
+            // Using default result when the 'path' doesn't exist
             return defaultResult;
         }
 
+        JsonNode evaluatedNode = applyPreoperation(context, nodeAtPath);
         return evaluate(context, path, evaluatedNode);
     }
 
-    private JsonNode applyPreoperation(ExpressionEvaluationContext globalContext) {
-        JsonNode payload = globalContext.getNode().at(path);
-
+    private JsonNode applyPreoperation(ExpressionEvaluationContext globalContext, JsonNode nodeAtPath) {
         if (null == preoperation) {
-            return payload;
+            return nodeAtPath;
         }
 
         ExpressionEvaluationContext nodeEvaluationContext = globalContext.deepCopy();
-        nodeEvaluationContext.setNode(payload);
+        nodeEvaluationContext.setNode(nodeAtPath);
 
         val computedValue = preoperation.compute(nodeEvaluationContext);
-        payload = mapper.valueToTree(computedValue);
-
-        return payload;
+        return mapper.valueToTree(computedValue);
     }
 
     abstract protected boolean evaluate(ExpressionEvaluationContext context, final String path, JsonNode evaluatedNode);
+
+    private static final class JacksonConfiguration implements Configuration.Defaults {
+        private final JsonProvider jsonProvider = new JacksonJsonProvider();
+        private final MappingProvider mappingProvider = new JacksonMappingProvider();
```

Comment:
Does this create another object mapper?

Reply:
No, this is to tell the JsonPath to use JacksonJsonProvider (jackson-databind) through the SPI provided. Default is JsonSmartJsonProvider.